public/pages/Main/Main.tsx

```diff
@@ -257,6 +260,11 @@ export default class Main extends Component<MainProps, MainState> {
     }
   };
 
+  dataSourceFilterFn = (dataSource: SavedObject<DataSourceAttributes>) => {
+    const engineVersion = dataSource?.attributes?.dataSourceVersion || "";
```

Comment:
Besides `engineVersion` check we may also need to check if `requiredOSDataSourcePlugins` are installed on the data source as well.  The installed plugin metadata is available as `dataSource?.attributes?.installedPlugins` which would be an array containing all installed backed plugins on the datasource.

Maybe some check like 
`pluginManifest.requiredOSDataSourcePlugins.every((plugin) => dataSource?.attributes?installedplugins?.includes(plugin))`

Reply:
Ack, addressed as part of next commit.